exercises/concept/hyperia-forex/HyperiaForexTests.cs

```diff
@@ -17,8 +17,117 @@ public void Equality_false()
     }
 
     [Fact(Skip = "Remove this Skip property to run this test")]
-    public void Equality_bad()
+    public void Equality_different_currencies()
     {
         Assert.Throws<ArgumentException>(() => new CurrencyAmount(55, "HD") == new CurrencyAmount(60, "USD"));
     }
+
+    [Fact(Skip = "Remove this Skip property to run this test")]
+    public void Inequality_true()
+    {
+        Assert.True(new CurrencyAmount(55, "HD") != new CurrencyAmount(60, "HD"));
+    }
+
+    [Fact(Skip = "Remove this Skip property to run this test")]
+    public void Inequality_false()
+    {
+        Assert.False(new CurrencyAmount(55, "HD") != new CurrencyAmount(55, "HD"));
+    }
+
+    [Fact(Skip = "Remove this Skip property to run this test")]
+    public void Inequality_different_currencies()
+    {
+        Assert.Throws<ArgumentException>(() => new CurrencyAmount(55, "HD") != new CurrencyAmount(60, "USD"));
+    }
+
+    [Fact(Skip = "Remove this Skip property to run this test")]
+    public void LessThan_true()
+    {
+        Assert.True(new CurrencyAmount(55, "HD") < new CurrencyAmount(60, "HD"));
+    }
+
+    [Fact(Skip = "Remove this Skip property to run this test")]
+    public void LessThan_false()
+    {
+        Assert.False(new CurrencyAmount(60, "HD") < new CurrencyAmount(55, "HD"));
+    }
+
+    [Fact(Skip = "Remove this Skip property to run this test")]
+    public void LessThan_different_currencies()
+    {
+        Assert.Throws<ArgumentException>(() => new CurrencyAmount(55, "HD") < new CurrencyAmount(60, "USD"));
+    }
+
+    [Fact(Skip = "Remove this Skip property to run this test")]
+    public void GreaterThan_true()
+    {
+        Assert.True(new CurrencyAmount(60, "HD") > new CurrencyAmount(55, "HD"));
+    }
+
+    [Fact(Skip = "Remove this Skip property to run this test")]
+    public void GreaterThan_false()
+    {
+        Assert.False(new CurrencyAmount(55, "HD") > new CurrencyAmount(60, "HD"));
+    }
+
+    [Fact(Skip = "Remove this Skip property to run this test")]
+    public void GreaterThan_different_currencies()
+    {
+        Assert.Throws<ArgumentException>(() => new CurrencyAmount(60, "HD") > new CurrencyAmount(55, "USD"));
+    }
+
+    [Fact(Skip = "Remove this Skip property to run this test")]
+    public void Addition_same_currencies()
+    {
+        Assert.Equal(new CurrencyAmount(155, "HD"), new CurrencyAmount(55, "HD") + new CurrencyAmount(100, "HD"));
```

Comment:
Note that these test cases are very basic and only cover about as much as is shown in the instructions. I feel like this exercise is a perfect candidate to use property-based testing instead, so to test the `+` operator overload happy cases for instance, we could test:

- Commutativity: `x + y = y + x`
- Associativity: `(x + y) + z = x + (y + z)`

IMO these tests do a better job of verifying the exercise implementation and may even trigger students to think about edge-cases they did not consider yet. 

I'm not sure if property-based testing is used at all in Exercism test suites, but it could be worth considering to do in the future. What do you think?

Reply:
I gave it a shot, have a look to see if you think it's still readable and understandable for students.